packages/core/src/common/message-rpc/rpc-protocol.ts

```diff
@@ -145,21 +144,20 @@ export class RpcProtocol {
     }
 
     sendRequest<T>(method: string, args: any[]): Promise<T> {
-        const id = this.nextMessageId++;
-        const reply = new Deferred<T>();
-
         // The last element of the request args might be a cancellation token. As these tokens are not serializable we have to remove it from the
         // args array and the `CANCELLATION_TOKEN_KEY` string instead.
         const cancellationToken: CancellationToken | undefined = args.length && CancellationToken.is(args[args.length - 1]) ? args.pop() : undefined;
         if (cancellationToken && cancellationToken.isCancellationRequested) {
             return Promise.reject(this.cancelError());
         }
 
+        const id = this.nextMessageId++;
+        const reply = new Deferred<T>();
+
         if (cancellationToken) {
             args.push(RpcProtocol.CANCELLATION_TOKEN_KEY);
             cancellationToken.onCancellationRequested(() => {
                 this.sendCancel(id);
-                this.pendingRequests.get(id)?.reject(this.cancelError());
```

Comment:
So the idea is that the "other side" will cancel the call, and send a "replyCancel"?

Reply:
To my understanding the idea here is that the request sender (client) just informs he request handler (server) that it has cancelled the request and is no longer interested in the result. How the server reacts to this cancel notification is up to the actual method implementation. The implementation can either 
- Not listen to the cancellation notification at all and just complete normally
- React to the cancel notification and throw an error (Ideally but not necessarily  a `CancellationError`). 

At least that's the way it was implemented before this PR :wink: